code/modules/games/cards.dm

```diff
@@ -109,11 +121,59 @@
 	var/datum/playingcard/P = cards[1]
 	H.cards += P
 	cards -= P
-	H.parentdeck = src
+	H.parentdeck = src.decktype
 	H.update_icon()
 	user.visible_message("<span class='notice'>\The [user] draws a card.</span>")
 	to_chat(user,"<span class='notice'>It's the [P].</span>")
 
+/obj/item/weapon/deck/verb/find_card() /// A bit of copypasta code but I'm not sure how to avoid that.
+	set category = "Object"
+	set name = "Find Card"
+	set desc = "Find a specific card from a deck."
+	set src in view(1)
+
+	var/mob/living/carbon/user = usr
```

Comment:
You should definitely typecheck this before trying to use it, `usr` can be dangerous. Any particular reason for carbon and not human?